src/uu/ls/src/ls.rs

```diff
@@ -779,9 +824,39 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
                     options::CLASSIFY,
                     options::INDICATOR_STYLE,
                 ]))
+        .arg(
+            //This still needs support for posix-*, +FORMAT
+            Arg::with_name(options::TIME_STYLE)
+                .long(options::TIME_STYLE)
+                .help("time/date format with -l; see TIME_STYLE below")
+                .value_name("TIME_STYLE")
+                .possible_values(&[
+                    "full-iso",
+                    "long-iso",
+                    "iso",
+                    "locale",
+                ])
+                .overrides_with_all(&[
+                    options::TIME_STYLE,
+                ])
+        )
+        .arg(
+            Arg::with_name(options::FULL_TIME)
+            .long(options::FULL_TIME)
+            .help("like -l --time-style=full-iso")
+            .overrides_with_all(&[
+                options::TIME_STYLE,
+                options::format::LONG,
+            ])
```

Comment:
> I suggest looking at how -o, -g and -n are implemented
I already looked at those exact implementations but I don't think the issue is actually exactly the same for them.

> ls --full-time -l should still display the long iso format
That does work, but for instance ` --full-time --time-style=iso` doesn't display long options right now.

Reply:
Yeah `clap` can be quite tricky with these sort of multi-functional flag. Could you also check whether
```
ls --full-time -C
ls --full-time -Cl
```
show the columns format and the long format with `full-iso`, respectively?

> That does work, but for instance --full-time --time-style=iso doesn't display long options right now.

This is probably because `--full-time` has `--time-style` as an override, but removing that will probably result in one always being chosen over the other.

If I understand correctly this argument sort of requires you to do the strategy for `-o` twice, once for the format and once for the time-style, which means
- Don't add any overrides in `clap`.
- Get the index of the last "regular" format argument, compare with the last index of `--full-time`. If the of the `--full-time` index is greater set the format to `long`.
- Get the index of the last "regular" time-style argument, compare with the last index of `--full-time`. If the index of `--full-time` is greater, set the time-style to `full-iso`.

That should probably work right? However, I find this method a bit weird, so it would be great if you find a better method.